pkg/user/tx_client_test.go

```diff
@@ -447,3 +452,241 @@ func setupEstimatorService(t *testing.T) *mockEstimatorServer {
 	t.Cleanup(mes.stop)
 	return mes
 }
+
+var (
+	errMock1             = errors.New("mock1 failed")
+	errMock2             = errors.New("mock2 failed")
+	errMock3             = errors.New("mock3 failed")
+	errInsufficientFunds = errors.New("insufficient funds") // Replicates SDK error text
+)
+
+type broadcastTestCase struct {
+	setupMocks  func(t *testing.T) ([]*grpctest.MockTxService, []*grpc.ClientConn, []func())
+	expectedErr error
+}
+
+func (suite *TxClientTestSuite) TestBroadcastScenarios() {
+	t := suite.T()
+
+	// Default options for most tests - used only to create a valid tx.
+	defaultOpts := []user.TxOption{user.SetGasLimit(100000), user.SetFee(1000)}
+	// Basic MsgSend for testing - use the main suite's default address.
+	defaultMsg := bank.NewMsgSend(suite.txClient.DefaultAddress(), suite.txClient.DefaultAddress(), sdk.NewCoins(sdk.NewCoin(appconsts.BondDenom, sdkmath.NewInt(10))))
+
+	testCases := []broadcastTestCase{
+		{ // Primary Success (Single Conn)
+			setupMocks: func(t *testing.T) ([]*grpctest.MockTxService, []*grpc.ClientConn, []func()) {
+				mockSvc1 := &grpctest.MockTxService{
+					BroadcastHandler: func(ctx context.Context, req *sdktx.BroadcastTxRequest) (*sdktx.BroadcastTxResponse, error) {
+						return &sdktx.BroadcastTxResponse{TxResponse: &sdk.TxResponse{Code: abci.CodeTypeOK, TxHash: "HASH1"}}, nil
+					},
+				}
+				conn1, stop1 := grpctest.StartMockServer(t, mockSvc1)
+				return []*grpctest.MockTxService{mockSvc1}, []*grpc.ClientConn{conn1}, []func(){stop1}
+			},
+			expectedErr: nil,
+		},
+		{ // Secondary Success
```

Comment:
using sleeps like this (ms) will definitely flake in CI - another way you can test is by allowing the second server to return immediately (or with small delay), and the rest with longer delays.

Reply:
done